backend/src/ml_space_lambda/app_configuration/policy_helper/notebook.py

```diff
@@ -0,0 +1,90 @@
+#
+#   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+#   Licensed under the Apache License, Version 2.0 (the "License").
+#   You may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+import json
+import logging
+
+import boto3
+
+from ml_space_lambda.utils.iam_manager import IAMManager
+from ml_space_lambda.utils.mlspace_config import get_environment_variables, retry_config
+
+iam = boto3.client("iam", config=retry_config)
+log = logging.getLogger(__name__)
+
+
+def create_instance_constraint_statement(actions: list[str], resources: list[str], allowed_instances: list[str]):
+    return {
+        "Effect": "Allow",
+        "Action": actions,
+        "Resource": resources,
+        "Condition": {"ForAnyValue:StringEquals": {"sagemaker:InstanceTypes": allowed_instances}},
+    }
+
+
+def create_sagemaker_resource_arn(resource: str, context) -> str:
+    components = context.invoked_function_arn.split(":")
+    return f"arn:{components[1]}:sagemaker:{components[3]}:{components[4]}:{resource}/*"
+
+
+def update_instance_constraint_policies(new_configuration, context) -> None:
```

Comment:
in this function I see it handles:

* Training/HPO
* Transform
* Endpoints

Do we need to handle Notebooks here too? That would be the last configurable service with instance types

Reply:
I'll check but I thought we don't allow notebooks to launch other notebooks.

Reply:
Ah okay I missed that detail, I thought this was app-wide, not specific to notebooks. This looks good then